modules/idle-scheduler/src/lib.rs

```diff
@@ -77,13 +87,51 @@ pub mod module {
 	#[pallet::getter(fn next_task_id)]
 	pub type NextTaskId<T: Config> = StorageValue<_, Nonce, ValueQuery>;
 
+	///
+	#[pallet::storage]
+	#[pallet::getter(fn previous_relay_block)]
+	pub type PreviousRelayBlockNumber<T: Config> = StorageValue<_, BlockNumber, ValueQuery>;
+
 	#[pallet::pallet]
 	pub struct Pallet<T>(_);
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(_n: T::BlockNumber) -> Weight {
+			// Unwrap shouldn't fail if it does defaults to 0
+			let test: BlockNumber = T::RelayChainBlockNumberProvider::current_block_number()
```

Comment:
```suggestion
			let prev_relay: BlockNumber = T::RelayChainBlockNumberProvider::current_block_number()
```

And add a comment explain this is actually the previous relay chain block because `on_initialize` is executed before inherent are executed, which sets the new relay chain block number